presto-main/src/main/java/com/facebook/presto/sql/planner/plan/GroupIdNode.java

```diff
@@ -36,30 +37,33 @@
         extends PlanNode
 {
     private final PlanNode source;
-    private final List<List<Symbol>> groupingSets;
-    private final Map<Symbol, Symbol> identityMappings;
+    private final List<Map<Symbol, Symbol>> groupingSetMappings;
+    private final Map<Symbol, Symbol> argumentMappings;
     private final Symbol groupIdSymbol;
 
     @JsonCreator
     public GroupIdNode(@JsonProperty("id") PlanNodeId id,
             @JsonProperty("source") PlanNode source,
-            @JsonProperty("groupingSets") List<List<Symbol>> groupingSets,
-            @JsonProperty("identityMappings") Map<Symbol, Symbol> identityMappings,
+            @JsonProperty("groupingSetMappings") List<Map<Symbol, Symbol>> groupingSetMappings,
+            @JsonProperty("argumentMappings") Map<Symbol, Symbol> argumentMappings,
             @JsonProperty("groupIdSymbol") Symbol groupIdSymbol)
     {
         super(id);
         this.source = requireNonNull(source);
-        this.groupingSets = ImmutableList.copyOf(requireNonNull(groupingSets));
-        this.identityMappings = ImmutableMap.copyOf(requireNonNull(identityMappings));
+        this.groupingSetMappings = ImmutableList.copyOf(requireNonNull(groupingSetMappings));
+        this.argumentMappings = ImmutableMap.copyOf(requireNonNull(argumentMappings));
```

Comment:
Add a check to ensure that the keys in groupSetMappings don't overlap with those of argumentMappings.